src/Commands/Admin/SetSiteArchiveState.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.Online.SharePoint.TenantAdministration;
+using Microsoft.SharePoint.Client;
+using PnP.PowerShell.Commands.Base;
+using PnP.PowerShell.Commands.Base.PipeBinds;
+using PnP.PowerShell.Commands.Enums;
+using System;
+using System.Management.Automation;
+
+namespace PnP.PowerShell.Commands.Admin
+{
+    [Cmdlet(VerbsCommon.Set, "PnPSiteArchiveState")]
+    public class SetSiteArchiveState : PnPAdminCmdlet
+    {
+        [Parameter(Mandatory = false, Position = 0, ValueFromPipeline = true)]
+        public SPOSitePipeBind Identity;
+
+        [Parameter(Mandatory = true, Position = 1)]
+        public SPOArchiveState ArchiveState;
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter NoWait;
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter Force;
+
+        protected override void ExecuteCmdlet()
+        {
+            if (!ParameterSpecified(nameof(Identity)) || string.IsNullOrEmpty(Identity.Url))
+            {
+                throw new ArgumentNullException($"{nameof(Identity)} must be provided when setting {nameof(ArchiveState)} for a Site");
```

Comment:
Why is the Identity parameter at line 14/15 not required if the first thing you do is ensure it has been provided? Specific reason for doing it like this?

Reply:
Hmm, we can make it required !